src/components/AdminSettings.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+	<div id="deepl_prefs" class="section">
+		<h2>
+			<DeeplIcon class="icon" />
+			{{ t('integration_deepl', 'DeepL integration') }}
+		</h2>
+		<p class="settings-hint">
+			<span>
+				{{ t('integration_deepl', 'Set an API key to start using the DeepL integration') }}
+			</span>
+		</p>
+		<div id="deepl-content">
+			<div class="line">
+				<label for="deepl-apikey">
+					<KeyIcon :size="20" class="icon" />
+					{{ t('integration_deepl', 'API Key') }}
+				</label>
+				<input id="deepl-apikey"
+					v-model="state.apikey"
+					type="password"
+					:readonly="readonly"
+					:placeholder="t('integration_deepl', 'API key for DeepL')"
+					@input="onInput"
+					@focus="readonly = false">
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import KeyIcon from 'vue-material-design-icons/Key.vue'
+
+import { loadState } from '@nextcloud/initial-state'
+import { generateUrl } from '@nextcloud/router'
+import axios from '@nextcloud/axios'
+import { showSuccess, showError } from '@nextcloud/dialogs'
+import DeeplIcon from './icons/DeeplIcon.vue'
+
+let timeout
+const delay = (fn, ms = 2000) => {
+	clearTimeout(timeout)
+	timeout = setTimeout(fn, ms)
+}
+
+export default {
+	name: 'AdminSettings',
+
+	components: {
+		KeyIcon,
+		DeeplIcon,
+	},
+
+	props: [],
+
+	data() {
+		return {
+			readonly: true,
+			state: loadState('integration_deepl', 'admin-config'),
+		}
+	},
+
+	watch: {
+	},
+
+	mounted() {
+	},
+
+	methods: {
+		onInput() {
+			delay(() => {
+				this.saveOptions({
+					apikey: this.state.apikey,
+				})
+			})
+		},
+		saveOptions(values) {
+			const req = { values }
+			const url = generateUrl('/apps/integration_deepl/admin-config')
+
+			axios.put(url, req).then(() => {
+				showSuccess(t('integration_deepl', 'DeepL admin options saved'))
+			}).catch((error) => {
+				showError(
+					t('integration_deepl', 'Failed to save DeepL admin options')
+					+ ': ' + (error.response?.request?.responseText ?? '')
+				)
+				console.error(error)
+			})
+		},
```

Comment:
Personal preference would be async/await here

```suggestion
		async saveOptions(values) {
			const req = { values }
			const url = generateUrl('/apps/integration_deepl/admin-config')

			try {
				await axios.put(url, req)
				showSuccess(t('integration_deepl', 'DeepL admin options saved'))
			} catch {
				showError(
					t('integration_deepl', 'Failed to save DeepL admin options')
					+ ': ' + (error.response?.request?.responseText ?? '')
				)
				console.error(error)
			}
		},
```